source/Halibut/HalibutRuntimeBuilder.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Security.Cryptography.X509Certificates;
+using Halibut.Diagnostics;
+using Halibut.ServiceModel;
+
+namespace Halibut
+{
+    public class HalibutRuntimeBuilder
+    {
+        ILogFactory logFactory = new LogFactory();
+        IPendingRequestQueueFactory queueFactory;
+        X509Certificate2 serverCertificate;
+        IServiceFactory serviceFactory = new NullServiceFactory();
+        ITrustProvider trustProvider = new DefaultTrustProvider();
```

Comment:
Can we use `??=` to initialise these in the `Build` method rather than creating instances we potentially don't need? This will also discourage other builder methods from messing with instance fields which may be subsequently discarded.